integration_tests/src/main/python/array_test.py

```diff
@@ -625,3 +625,35 @@ def test_arrays_overlap_before_spark313(data_gen):
             'arrays_overlap(array(3, 4), array(1, 2))',
             'arrays_overlap(array(), array(1, 2))')
     )
+
+@pytest.mark.parametrize('data_gen', [byte_gen, short_gen, int_gen, long_gen], ids=idfn)
+def test_array_remove_scalar(data_gen):
+    gen = StructGen(
+        [('a', ArrayGen(data_gen, nullable=True))],
+        nullable=False)
+
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, gen).selectExpr(
+            'array_remove(a, -10)',
+            'array_remove(a, 0)',
+            'array_remove(a, 10)')
+    )
+
+@pytest.mark.parametrize('data_gen', [byte_gen, short_gen, int_gen, long_gen,
+                                      float_gen, double_gen,
+                                      string_gen, boolean_gen, date_gen, timestamp_gen] + decimal_gens, ids=idfn)
+def test_array_remove(data_gen):
+    gen = StructGen(
+        [('a', ArrayGen(data_gen, nullable=True)),
+        ('b', data_gen)],
```

Comment:
The probability on a per row basis of `b` being inside `a` is very small once we get past byte_gen. It would be nice if we added in some high probability special cases in most of these gens so there is a likelihood that we will actually end up removing things from the arrays.

```
[ByteGen(special_cases=[5]), ShortGen(special_cases=[5]),...]
```